File: beginner/__3000__/3507/3507.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    long long number_animais, number_legs; 
    cin >> number_animais >> number_legs;

    // TOTAL DE ANIMAIS = NUMERO DE GANSOS + NUMERO DE TIGRES
    // NUMERO DE TIGRES = TOTAL DE ANIMAIS - NUMERO DE GANSOS

    // LEGS = (2 * GANSOS) + (4 * TIGRES)
    // LEGS = (2 * GANSOS) + (4 * (N_ANIMAIS - GANSOS))
    // LEGS = 2*GANSOS + 4*N_ANIMAIS - 4*GANSOS
    // LEGS - 4*N_ANIMAIS = 2*GANSOS - 4*GANSOS
    // LEGS - 4*N_ANIMAIS = -2*GANSOS
    // GANSOS = (LEGS - (4 * N_ANIMAIS)) / -2

    long long G = abs(number_legs - (4 * number_animais)) / 2;
    long long T = number_animais - G;

    cout << T << "\n" << G << "\n";

    return 0;
}

File: beginner/__3000__/3507/3507.py
N = int(input())    # numero de animais
M = int(input())    # numero de patas

# M = (2 * G) + (4 * T)
# G == numero de gansos
# T == numero de tigres

# ENCONTRA O VALOR DE G ou T para termos apenas
# UMA VARIAVEL NA EQUAÇÃO.
# N = G + T 
# N - G = T
# N - T = G

# ENCONTRA O NUMERO DE GANSOS.
# M = (2 * G) + (4 * (N - G))
# M = 2G + 4N - 4G
# M - 4N = 2G - 4G
# G = |M|
G = int(abs((M - 4*N) / -2))

# ENCONTRA O NUMERO DE TIGRES.
# M = (2 * Gansos) * (4 * T)
# M / (2*G) = 4T
# T = |M|
T = int(abs((M - 2*G) / 4))

print(T)
print(G)